generic/tohil.c

```diff
@@ -122,6 +122,76 @@ tohil_UTF8ToTclDString(Tcl_Interp *interp, char *utf8String, int utf8StringLen,
     return Tcl_ExternalToUtfDString(utf8encoding, utf8String, utf8StringLen, ds);
 }
 
+//
+// tohil_UndentPython - remove consistent indenting from a python code block
+//
+static int tohil_UndentPython(Tcl_Interp *interp, char *string) {
+    char *indent = malloc(strlen(string)+1);
+    char *code_ptr = string;
+    char *indent_ptr = indent;
+    int seen_code = 0;
+
+    // look for indent
+    while (*code_ptr) {
+        // skip blank lines;
+        if(*code_ptr == '\n') {
+            ++code_ptr;
+            indent_ptr = indent;
+            continue;
+        }
+        // Got some code, we have an indent
+        if (!isspace(*code_ptr)) {
+            seen_code = 1;
+            *indent_ptr = '\0';
+            break;
+        }
+        // Copy blanks into indent;
+        *indent_ptr++ = *code_ptr++;
+    }
+
+    // Empty code block, so just pass it back unchanged.
+    if(!seen_code) {
+        free(indent);
+        return TCL_OK;
+    }
+
+    // walk string deleting indent on every line
+    //
+    // It has already been deleted and copied into the indent buffer
+    // for the first line, so we only start the check when we see a
+    // new newline.
+    //
+    char *working_ptr = string;
+    while(*code_ptr) {
+        char c = *working_ptr++ = *code_ptr++;
+        if(c == '\n') {
+            indent_ptr = indent;
+            while(*indent_ptr && *code_ptr) {
+                // Look for mismatches
+                if(*indent_ptr != *code_ptr) {
+                    // Fast forward to end of line
+                    while(*code_ptr != '\n' && isspace(*code_ptr))
+                        code_ptr++;
+                    // Mismatch on blank line, ignore
+                    if(*code_ptr == '\n')
```

Comment:
Would it be useful to also check for `!*code_ptr` here (end of string), in case the last line of the code was a line of just a portion of the whitespace in ident? Or do you wish that to also be treated as an error?

Reply:
Good catch, I had that handled in an earlier version of the code but realized that a mismatched blank line that was longer than the undent should also be ignores, and lost the simpler case.